### PR TITLE
Add variables bindings and genericInformation to selection scripts #2565

### PR DESCRIPTION
**Pull request by fviale**
- add bindings to the Criteria class (which already contain the selection scripts list)
- add the bindings parameter to the RMNode interface (called on the node from the rm server)
- changed ScriptExecutor behavior to display the selection script output inside the rm logs by default
- in the SchedulingMethodImpl fill, the bindings with the task and job info, generic information, propagated variables if they are available or default job variables
- added a test which verifies that the bindings are correctly set

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java [hunk 1]
```diff
@@ -460,6 +467,53 @@ protected NodeSet getRMNodes(Map<JobId, JobDescriptor> jobMap, int neededResourc
...
+            if ((resultSize > 0) && task.handleResultsArguments()) {
+                List<TaskId> parentIds = new ArrayList<>(resultSize);
+                for (int i = 0; i < resultSize; i++) {
+                    parentIds.add(etd.getParents().get(i).getTaskId());
+                }
+                Map<TaskId, TaskResult> taskResults = schedulingService.getInfrastructure().getDBManager()
+                        .loadTasksResults(
+                                job.getId(), parentIds);
+                for (TaskResult taskResult : taskResults.values()) {
+                    if (taskResult.getPropagatedVariables() != null) {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java]: MAJOR Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS134)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by lpellegr** [comment]
Generic Information defined at the task level only are available from selection scripts. It would be nice to also import Generic information from the Job level. If there is a conflict between some defined at the task level, these last should probablyt have priority.

Please squash or rename commits with same names.

It would be also nice to update the documentation to reference the new genericInformation binding and the usage of variables from selection scripts.

**Comment by fviale** [comment]
all agreeing. Regarding documentation, that was the plan. I need to first add genericInformation bindings to other relevant scripts and then I will do a documentation pull request

**Pull request merged**